iidm/iidm-extensions/src/main/java/com/powsybl/iidm/network/extensions/SlackBusAdderImpl.java

```diff
@@ -0,0 +1,59 @@
+/**
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.iidm.network.extensions;
+
+import com.powsybl.commons.PowsyblException;
+import com.powsybl.commons.extensions.AbstractExtensionAdder;
+import com.powsybl.iidm.network.Terminal;
+import com.powsybl.iidm.network.VoltageLevel;
+
+import java.util.OptionalInt;
+
+/**
+ * @author Florian Dupuy <florian.dupuy at rte-france.com>
+ */
+public class SlackBusAdderImpl extends AbstractExtensionAdder<VoltageLevel, SlackBus> implements SlackBusAdder {
+
+    private String busId;
+    private OptionalInt node;
```

Comment:
Optional as a class attribute is often not recommended (https://stackoverflow.com/questions/29033518/is-it-a-good-practice-to-use-optional-as-an-attribute-in-a-class)